homeassistant/components/snooz/config_flow.py

```diff
@@ -0,0 +1,206 @@
+"""Config flow for Snooz component."""
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+from typing import Any
+
+from pysnooz.advertisement import SnoozAdvertisementData
+import voluptuous as vol
+
+from homeassistant.components.bluetooth import (
+    BluetoothScanningMode,
+    BluetoothServiceInfo,
+    async_discovered_service_info,
+    async_process_advertisements,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_ADDRESS, CONF_NAME, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+# number of seconds to wait for a device to be put in pairing mode
+WAIT_FOR_PAIRING_TIMEOUT = 30
+
+
+@dataclass
+class DiscoveredSnooz:
+    """Represents a discovered Snooz device."""
+
+    info: BluetoothServiceInfo
+    device: SnoozAdvertisementData
+
+
+class SnoozConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Snooz."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovery: DiscoveredSnooz | None = None
+        self._discovered_devices: dict[str, DiscoveredSnooz] = {}
+        self._pairing_task: asyncio.Task | None = None
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        device = SnoozAdvertisementData()
+        if not device.supported(discovery_info):
+            return self.async_abort(reason="not_supported")
+        self._discovery = DiscoveredSnooz(discovery_info, device)
+        return await self.async_step_bluetooth_confirm()
+
+    async def async_step_bluetooth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm discovery."""
+        assert self._discovery is not None
+
+        if user_input is not None:
+            if not self._discovery.device.is_pairing:
+                return await self.async_step_wait_for_pairing_mode()
+
+            return self._create_snooz_entry(self._discovery)
+
+        self._set_confirm_only()
+        assert self._discovery.device.display_name
+        placeholders = {"name": self._discovery.device.display_name}
+        self.context["title_placeholders"] = placeholders
+        return self.async_show_form(
+            step_id="bluetooth_confirm", description_placeholders=placeholders
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step to pick discovered device."""
+        if user_input is not None:
+            name = user_input[CONF_NAME]
+
+            discovered = self._discovered_devices.get(name)
```

Comment:
Please don't use `dict.get` if we know that the key is in the dict. Use `dict[key]`.